src/core/xlio_extra.h

```diff
@@ -641,4 +645,163 @@ static inline struct xlio_api_t *xlio_get_api()
     return api_ptr;
 }
 
+extern "C" {
+int xlio_socket(int __domain, int __type, int __protocol);
+
+int xlio_close(int __fd);
+
+void xlio___res_iclose(res_state statp, bool free_addr);
+
+int xlio_shutdown(int __fd, int __how);
+
+int xlio_listen(int __fd, int backlog);
+
+int xlio_accept(int __fd, struct sockaddr *__addr, socklen_t *__addrlen);
+
+int xlio_accept4(int __fd, struct sockaddr *__addr, socklen_t *__addrlen, int __flags);
+
+int xlio_bind(int __fd, const struct sockaddr *__addr, socklen_t __addrlen);
+
+int xlio_connect(int __fd, const struct sockaddr *__to, socklen_t __tolen);
+
+int xlio_setsockopt(int __fd, int __level, int __optname, __const void *__optval,
+                    socklen_t __optlen);
+
+int xlio_getsockopt(int __fd, int __level, int __optname, void *__optval, socklen_t *__optlen);
+
+int xlio_fcntl(int __fd, int __cmd, ...);
+
+int xlio_fcntl64(int __fd, int __cmd, ...);
+
+int xlio_ioctl(int __fd, unsigned long int __request, ...);
+
+int xlio_getsockname(int __fd, struct sockaddr *__name, socklen_t *__namelen);
+
+int xlio_getpeername(int __fd, struct sockaddr *__name, socklen_t *__namelen);
+
+ssize_t xlio_read(int __fd, void *__buf, size_t __nbytes);
+
+#if defined HAVE___READ_CHK
+ssize_t xlio___read_chk(int __fd, void *__buf, size_t __nbytes, size_t __buflen);
+#endif
+
+ssize_t xlio_readv(int __fd, const struct iovec *iov, int iovcnt);
+
+ssize_t xlio_recv(int __fd, void *__buf, size_t __nbytes, int __flags);
+
+#if defined HAVE___RECV_CHK
+ssize_t xlio___recv_chk(int __fd, void *__buf, size_t __nbytes, size_t __buflen, int __flags);
+#endif
+
+ssize_t xlio_recvmsg(int __fd, struct msghdr *__msg, int __flags);
+
+#ifndef HAVE_STRUCT_MMSGHDR
+#ifndef __INTEL_COMPILER
+struct mmsghdr {
+    struct msghdr msg_hdr; // Message header
+    unsigned int msg_len; // Number of received bytes for header
+};
+#endif
+#endif
+
+#ifndef MSG_WAITFORONE
+#define MSG_WAITFORONE 0x10000 // recvmmsg(): block until 1+ packets avail
+#endif
+
+#ifdef RECVMMSG_WITH_CONST_TIMESPEC
+int xlio_recvmmsg(int __fd, struct mmsghdr *__mmsghdr, unsigned int __vlen, int __flags,
+                  const struct timespec *__timeout);
+#else
+int xlio_recvmmsg(int __fd, struct mmsghdr *__mmsghdr, unsigned int __vlen, int __flags,
+                  struct timespec *__timeout);
+#endif
```

Comment:
I'd prefer to keep a single prototype - user uses xlio extra API explicitly what doesn't have to follow specific libc implementation byte by byte. const version looks like version which will work in more cases.

Also, having such XLIO specific define in public header is error prone. AFAIK we don't expose this define to pkg-config file and there is no guarantee that user will use pkg-config.

Reply:
The two variations here are to support Linux (no const) and BSD (const). It shouldn't be selectable by the user - it should be determined by the build system. I think this has to remain here if XLIO supports BSD.

Reply:
I will leave the const variant since this is a new symbol.